verl/utils/transferqueue_utils.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2025 Bytedance Ltd. and/or its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import inspect
+import threading
+from functools import wraps
+from typing import Any, Callable
+
+from tensordict import TensorDict
+
+try:
+    from transfer_queue import (
+        AsyncTransferQueueClient,
+        BatchMeta,
+        ZMQServerInfo,
+    )
+
+    HAS_TQ = True
```

Comment:
We can add TransferQueue option in `ppo_trainer.yaml`, if TransferQueue is enabled in config file, we can enable it globally by ray runtime environment variable, e.g `TRANSFER_QUEUE_ENABLE: 1`.
https://github.com/volcengine/verl/blob/main/verl/trainer/main_ppo.py#L66